### PR TITLE
Implement: Creature:SetRegeneratingHealth

### PR DESCRIPTION
**Pull request by xIvan0ff**
Implements the setter Creature:SetRegeneratingHealth

### CODE REVIEW

**Comment by Rochet2** [comment]
Doesnt seem like such method exists on mangos though as the status comes from creature info (static data in DB).
https://github.com/mangoszero/server/blob/e45aa74e0fa3a9677d90fc012b8161e1e9935742/src/game/Object/Creature.h#L806
The function should probably be restricted to AC and TC only.
Also, the extra changes to the code style should be removed.

**Comment by xIvan0ff** [comment]
> 
> 
> Doesnt seem like such method exists on mangos though as the status comes from creature info (static data in DB).
> https://github.com/mangoszero/server/blob/e45aa74e0fa3a9677d90fc012b8161e1e9935742/src/game/Object/Creature.h#L806
> The function should probably be restricted to AC and TC only.
> Also, the extra changes to the code style should be removed.

What kind of changes?

File: CreatureMethods.h [hunk 1]
```diff
@@ -73,10 +90,10 @@ namespace LuaCreature
      * @param bool checkFaction = true : if `true`, the [Creature] must be the same faction as `friend` to assist
      * @return bool canAssist
      */
-    int CanAssistTo(lua_State* L, Creature* creature)
+    int CanAssistTo(lua_State *L, Creature *creature)
     {
-        Unit* u = Eluna::CHECKOBJ<Unit>(L, 2);
-        Unit* enemy = Eluna::CHECKOBJ<Unit>(L, 3);
+        Unit *u = Eluna::CHECKOBJ<Unit>(L, 2);
+        Unit *enemy = Eluna::CHECKOBJ<Unit>(L, 3);
```
Comment (Rochet2) [inline CreatureMethods.h:96]: For example these

Comment (xIvan0ff) [reply CreatureMethods.h]: Fixed them, was due to automatic code formatting

**Comment by xIvan0ff** [comment]
There we go, added them :)

**Comment by xIvan0ff** [comment]
> 
> 
> Doesnt seem like such method exists on mangos though as the status comes from creature info (static data in DB).
> https://github.com/mangoszero/server/blob/e45aa74e0fa3a9677d90fc012b8161e1e9935742/src/game/Object/Creature.h#L806
> The function should probably be restricted to AC and TC only.
> Also, the extra changes to the code style should be removed.

Also, how do I restrict it to AC/TC only?

**Comment by Rochet2** [comment]
> Also, how do I restrict it to AC/TC only?

Take a look at these:
https://github.com/ElunaLuaEngine/Eluna/blob/fa6fc881e62fc17941f8dedf6cd6801366bf8a4d/LuaFunctions.cpp#L814-L818
https://github.com/ElunaLuaEngine/Eluna/blob/fa6fc881e62fc17941f8dedf6cd6801366bf8a4d/CreatureMethods.h#L278-L284

**Comment by xIvan0ff** [comment]
> 
> 
> > Also, how do I restrict it to AC/TC only?
> 
> Take a look at these:
> 
> https://github.com/ElunaLuaEngine/Eluna/blob/fa6fc881e62fc17941f8dedf6cd6801366bf8a4d/LuaFunctions.cpp#L814-L818
> 
> https://github.com/ElunaLuaEngine/Eluna/blob/fa6fc881e62fc17941f8dedf6cd6801366bf8a4d/CreatureMethods.h#L278-L284

There we go! :)
Also I was wondering does the Wiki update automatically?

**Comment by Rochet2** [comment]
> Also I was wondering does the Wiki update automatically?

A tool is used to autogenerate documentation from the comments in code.
But triggering the generation and publishing the generated doc online is manual.
It could be automated though. See https://github.com/ElunaLuaEngine/Eluna/blob/master/docs/DOC_GEN.md

**Comment by xIvan0ff** [comment]
Can you accept the PR <3 :(

**Comment by xIvan0ff** [comment]
Whoops... I guess that's what happens when you write code in Notepad :P thanks ! :)

**Pull request merged**